R-package/tests/testthat/test_parameters.R

```diff
@@ -101,14 +101,83 @@ test_that("training should warn if you use 'dart' boosting, specified with 'boos
         , num_threads = .LGB_MAX_THREADS
     )
     params[[boosting_param]] <- "dart"
+
+    # expect warning
+    # if early_stopping_rounds is specified
     expect_warning({
       result <- lightgbm(
         data = train$data
         , label = train$label
         , params = params
         , nrounds = 5L
         , verbose = -1L
+        , early_stopping_rounds = 1
       )
     }, regexp = "Early stopping is not available in 'dart' mode")
+
+    # expect no warning
+    # if early_stopping_rounds is not
+    # specified see:
+    # https://github.com/microsoft/LightGBM/issues/6612
```

Comment:
```suggestion
    # no warning: early stopping not requested
```

Please consolidate this to one line, and let's drop the link to the original issue. Anyone wanting to understand why this is the way it is can use the git blame to trace back to the pull request and issue (https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#viewing-the-line-by-line-revision-history-for-a-file).